dockerplugin.py

```diff
@@ -611,14 +714,43 @@ def read_callback(self):
                 if not read_at:
                     # No stats available yet; skipping container.
                     continue
-
                 # Process stats through each reader.
                 for method in self.METHODS:
                     method(container, cstats.dimensions, stats, read_at)
+
+                # If CPU shares or quota metrics are required
+                if self.cpu_shares_bool or self.cpu_quota_bool:
+                    # Get cgroup info container by inspecting the container
+                    container_inspect = self.client \
+                                                .inspect_container(container)
+                    containers_state[container['Name']] = {
```

Comment:
Why don't you index by `container` directly?

```python
containers_state[container] = container_inspect
```

If that doesn't work, then I don't think you need a dictionary here at all:

```python
containers_state = []
# ...
containers_state.append({
    'container': container,
    'state': container_inspect
})
```

Reply:
The first option will does not work because container being an object is not hashable. I'll make it a list instead of a dictionary.